File: transpiler/javatests/com/google/j2cl/transpiler/integration/subnativejstype/MyNativeType.js
goog.module('test.foo.MyNativeType');

class MyNativeType {
  constructor(x, y) {
    this.executed = true;
    this.x = x;
    this.y = y;
  }

  /**
   * @param {number} a
   * @return {number}
   */
  foo(a) { return this.x + this.y + a; }
}

exports = MyNativeType;

File: transpiler/javatests/com/google/j2cl/transpiler/integration/subnativejstype/MyNativeType.java
package com.google.j2cl.transpiler.integration.subnativejstype;

import jsinterop.annotations.JsType;

@JsType(isNative = true, namespace = "test.foo")
public class MyNativeType {
  MyNativeType(int x, int y) {}

  public boolean executed;
  public int x;
  public int y;

  public native int foo(int a);
}
